pemFioi/quickAlgo/interface-mobileFirst.js

```diff
@@ -302,7 +308,78 @@ var quickAlgoInterface = {
             this.displayBlocklyPython();
         } else if (btn == 'about') {
             this.openAbout();
+        } else if (btn == 'share') {
+            this.openShare();
+        }
+    },
+
+    openShare: function() {
+        displayHelper.showPopupMessage(this.strings.doYouWantToShare, 'blanket', this.strings.yes, this.shareSubject, this.strings.no, null, null)
+    },
+
+    shareSubject: function() {
+        // we take back quickAlgoInterface from global variable because the word "this" refer to displayHelper
+        var that = quickAlgoInterface;
+
+        var additional = that.userTaskData;
+
+        // This is done as if it was a task.js
+        var subTaskToPublish = {
+            // the title of the exercise
+            // we will have to move this to userTaskData during loading.
+            title: additional.title,
+
+            // taskIntro is the subject inside of the json editor.
+            // we will have to move this to userTaskData during loading.
+            taskIntro: additional.subject,
+
+            // This are the meta data of the subject, they have this format inside of the Json editor
+            // during load, we will take some information from here and put it in gridinfo (rebuild the about).
+            PEMTaskMetaData: {
+                // inside of the editor, authors is an array, in our case this is slightly different because it is a string,
+                // the owner of the task can write what he want there (website ect...)
+                // we will have to move this to userTaskData during loading.
+                authors: additional.about.authors,
+
+                // Normally this is how we set fr ect...
+                language: window.stringsLanguage,
+
+                // The user should be able to select his version during share, for now one, there is no way to display
+                // the version to the user, but we can set it so later if we need it (example if we want to create a
+                // versioning system), then all the subject already have their own version.
+                version: "1.0.0",
+
+                // the license of the subject, during load we will have to move it to the userTaskData and about.
+                license: additional.about.license
+            },
+
+            // We just need to take the gridInfo from the exercise.
+            gridInfos: that.subtask.gridInfos,
+
+            // We juste take the data from the exercise which mean that we are compatible with everything.
+            data: that.subtask.data,
+
+            // We retrieve the saved answer of the user.
+            // with this method, we can get the updated answer (after save) for all the levels.
+            answer: that.subtask.getAnswerObject()
```

Comment:
If we're going to share the answer, the user should be asked whether they want to share their answer or not. If I create my own task for others to try solving it, I don't want them to get my answer as well, or it would miss the point. On the other side, indeed, there are cases in which we'd want to share the program as well. Hence it's better to ask the user about their intent.